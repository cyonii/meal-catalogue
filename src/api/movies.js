import axios from 'axios';

const API_KEY = 'fca3a09ec5fa268a31aa58f3449d68be';
const BASE_URL = 'https://api.themoviedb.org/3/movie';

const fetchMoviesByType = async (type) => {
  const data = await axios.get(`${BASE_URL}/${type}?api_key=${API_KEY}&page=1`)
    .then((response) => response.data.results)
    .catch((error) => error.data);

  return data;
};

const fetchMovie = async (id) => {
  const data = axios.get(`${BASE_URL}/${id}?api_key=${API_KEY}`)
    .then((response) => response.data)
    .catch((error) => error.data);

  return data;
};

const fetchReviews = async (movieID) => {
  const data = axios.get(`${BASE_URL}/${movieID}/reviews?api_key=${API_KEY}`)
    .then((response) => response.data.results)
    .catch((error) => error.data);
  return data;
};

export { fetchMoviesByType, fetchMovie, fetchReviews };
