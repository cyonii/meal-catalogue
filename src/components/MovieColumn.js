import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';

const MovieColumn = (props) => {
  const { movie } = props;

  return (
    <Col className="col-6 col-sm-4 col-md-3">
      <Card className="movie-card bg-dark h-100 rounded-0 border-0">
        <Card.Img className="rounded-0" src={`https://image.tmdb.org/t/p/w780/${movie.poster_path}`} alt={movie.title} loading="lazy" />

        <Card.Body className="text-white opacity-100">
          <Card.Img className="card-img mb-2" src={`https://image.tmdb.org/t/p/w780/${movie.backdrop_path}`} />
          <Card.Title className="fs-6 fw-bold">{movie.title}</Card.Title>
          <Card.Text>
            <span className="text-muted">Rating:&nbsp;&nbsp;</span>
            <b>
              {movie.vote_average}
              &nbsp;/ 10
            </b>
          </Card.Text>
          <Link to={`/movie/${movie.id}`} className="btn btn-success border-success border-2 fw-bold">Movie Info</Link>
        </Card.Body>
      </Card>
    </Col>
  );
};

MovieColumn.propTypes = {
  movie: PropTypes.instanceOf(Object).isRequired,
};

export default MovieColumn;
