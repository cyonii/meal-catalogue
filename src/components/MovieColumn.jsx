import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';

const MovieColumn = (props) => {
  const { movie } = props;
  const assetUrl = 'https://image.tmdb.org/t/p/w780';
  const posterUrl = `${assetUrl}${movie.poster_path}`;

  return (
    <Col xs={6} sm={4} md={3} data-testid="movie-column">
      <Card className="movie-card bg-dark h-100 rounded-0 border-0">
        {/* Use placeholder if movie poster is unavailable */}
        { movie.poster_path
          ? <Card.Img className="rounded-0 bg-dark" src={posterUrl} alt={movie.title} loading="lazy" />
          : (
            <div className="movie-card-placeholder">
              <h6 className="text-light opacity-75">{movie.title}</h6>
              <i className="bi bi-image-alt display-4" />
            </div>
          )}

        <Card.Body className="text-white opacity-100">
          { movie.backdrop_path
            ? <Card.Img className="card-img mb-2" src={`https://image.tmdb.org/t/p/w780/${movie.backdrop_path}`} />
            : ''}
          <Card.Title className="fs-6 fw-bold">{movie.title}</Card.Title>
          <Card.Text>
            <span className="text-muted">Rating:&nbsp;&nbsp;</span>
            <b>{`${movie.vote_average} / 10`}</b>
          </Card.Text>
          <Link to={`/movie/${movie.id}`} className="btn btn-success border-success border-2 fw-bold">Movie Info</Link>
        </Card.Body>
      </Card>
    </Col>
  );
};

MovieColumn.propTypes = {
  movie: PropTypes.instanceOf(Object).isRequired,
};

export default MovieColumn;
