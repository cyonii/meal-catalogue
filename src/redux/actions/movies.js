import { ADD_MOVIES } from './types';

const addMovies = (movies) => ({
  type: ADD_MOVIES,
  payload: movies,
});

export { addMovies }; // eslint-disable-line
