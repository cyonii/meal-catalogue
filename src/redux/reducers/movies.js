import { ADD_MOVIES } from '../actions/types';

export default (state = [], action) => {
  switch (action.type) {
    case ADD_MOVIES:
      return action.payload;
    default:
      return state;
  }
};
