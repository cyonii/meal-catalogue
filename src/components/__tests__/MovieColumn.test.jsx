import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import MovieColumn from "../MovieColumn";
import movies from "./data/movies.json";

describe("MovieColumn", () => {
  it("renders without crashing", async () => {
    const movie = movies.results[1];

    const el = (
      <BrowserRouter>
        <MovieColumn movie={movie} />
      </BrowserRouter>
    );
    render(el);

    expect(screen.getByTestId("movie-column")).toMatchSnapshot();
    expect(screen.getByText(movie.title)).toBeInTheDocument();
    expect(screen.getByText("Movie Info")).toBeInTheDocument();
    expect(screen.getByText(`${movie.vote_average} / 10`)).toBeInTheDocument();
  });
});
