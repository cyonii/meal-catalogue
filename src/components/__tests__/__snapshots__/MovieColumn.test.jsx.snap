// Vitest Snapshot v1

exports[`MovieColumn > renders without crashing 1`] = `
<div
  class="col-md-3 col-sm-4 col-6"
  data-testid="movie-column"
>
  <div
    class="movie-card bg-dark h-100 rounded-0 border-0 card"
  >
    <img
      alt="Deadpool 2"
      class="card-img rounded-0 bg-dark"
      loading="lazy"
      src="https://image.tmdb.org/t/p/w780/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg"
    />
    <div
      class="text-white opacity-100 card-body"
    >
      <img
        class="card-img card-img mb-2"
        src="https://image.tmdb.org/t/p/w780//3P52oz9HPQWxcwHOwxtyrVV1LKi.jpg"
      />
      <div
        class="fs-6 fw-bold card-title h5"
      >
        Deadpool 2
      </div>
      <p
        class="card-text"
      >
        <span
          class="text-muted"
        >
          Rating:  
        </span>
        <b>
          7.6 / 10
        </b>
      </p>
      <a
        class="btn btn-success border-success border-2 fw-bold"
        href="/movie/383498"
      >
        Movie Info
      </a>
    </div>
  </div>
</div>
`;
